plugins/outputs/nats/nats.go

```diff
@@ -280,28 +319,66 @@ func (n *NATS) Write(metrics []telegraf.Metric) error {
 		return nil
 	}
 
+	// If the FieldName is included in the subject, we need to split the metric into multiple metrics
+	// each with a single field.
+	if n.includeFieldInSubject {
+		var newMetrics []telegraf.Metric
+		for _, m := range metrics {
+			newMetrics = append(newMetrics, splitMetricByField(m)...)
+		}
+		metrics = newMetrics
+	}
+
+	var bufSubject bytes.Buffer
```

Comment:
Please use short names...
```suggestion
	var subject bytes.Buffer
```

Reply:
Fixed